command/command.go

```diff
@@ -12,9 +12,68 @@ import (
 	"github.com/ernestio/ernest-cli/manager"
 	"github.com/ernestio/ernest-cli/model"
 	"github.com/urfave/cli"
+
+	emodels "github.com/ernestio/ernest-go-sdk/models"
+)
+
+const (
+	// NonEmptuTokenVal ...
+	NonEmptuTokenVal = "non_empty_token"
+	// NonAdminVal ...
+	NonAdminVal = "non_admin"
 )
 
+// NoValidation ...
+var NoValidation = []string{}
+
+// AuthUsersValidation ...
+var AuthUsersValidation = []string{NonEmptuTokenVal}
+
+// NonAdminValidation ...
+var NonAdminValidation = []string{NonEmptuTokenVal, NonAdminVal}
+
+type validation func(*manager.Client)
+
+var validations = map[string]validation{
+	NonEmptuTokenVal: func(client *manager.Client) {
+		if client.Config().Token == "" {
+			h.PrintError("You're not allowed to perform this action, please log in")
+		}
+	},
+	NonAdminVal: func(client *manager.Client) {
+		session := client.Session().Get()
+		if !session.IsAdmin() {
+			h.PrintError("You don’t have permissions to perform this action")
+		}
+	},
+}
+
+var session *emodels.Session
+
+// esetup ...
+func esetup(c *cli.Context, vals []string) *manager.Client {
```

Comment:
this method will be fully replacing [setup](https://github.com/ernestio/ernest-cli/pull/141/files#diff-f2f57f6e81c1ea7ce925abd60673525cR77) basically returning an manager.Client object instead of manager.Manager object.

Reply:
Additionally `esetup` function is accepting `vals` with an array of strings defining validations to be done on the input parameters.

This makes easy to share common code between different commands. Validations are defined [here](https://github.com/ernestio/ernest-cli/pull/141/files#diff-f2f57f6e81c1ea7ce925abd60673525cR37)